ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -1,225 +1,32 @@
-        const chainId = stringChainKey as Hex;
-        tokens.forEach((token: Token) => {
-          const { isNative, address, decimals } = token;
-          const balance =
-            calculateTokenBalance({
-              isNative,
-              chainId,
-              address,
-              decimals,
-              nativeBalances,
-              selectedAccountTokenBalancesAcrossChains,
-            }) || '0';
-
-          const tokenFiatAmount = calculateTokenFiatAmount({
-            token,
-            chainId,
-            balance,
-            marketData,
-            currencyRates,
-          });
-
-          // Respect the "hide zero balance" setting (when true):
-          // - Native tokens should always display with zero balance when on the current network filter.
-          // - Native tokens should not display with zero balance when on all networks filter
-          // - ERC20 tokens with zero balances should respect the setting on both the current and all networks.
-
-          // Respect the "hide zero balance" setting (when false):
-          // - Native tokens should always display with zero balance when on the current network filter.
-          // - Native tokens should always display with zero balance when on all networks filter
-          // - ERC20 tokens always display with zero balance on both the current and all networks filter.
-          if (
-            !hideZeroBalanceTokens ||
-            balance !== '0' ||
-            (token.isNative && isOnCurrentNetwork)
-          ) {
-            tokensWithBalance.push({
-              ...token,
-              balance,
-              tokenFiatAmount,
-              chainId,
-              string: String(balance),
-            });
-          }
-        });
-      },
-    );
-
-    return tokensWithBalance;
-  };
-
-  const sortedFilteredTokens = useMemo(() => {
-    const consolidatedTokensWithBalances = consolidatedBalances();
-    const filteredAssets = filterAssets(consolidatedTokensWithBalances, [
-      {
-        key: 'chainId',
-        opts: tokenNetworkFilter,
-        filterCallback: 'inclusive',
-      },
-    ]);
-
-    const { nativeTokens, nonNativeTokens } = filteredAssets.reduce<{
-      nativeTokens: TokenWithFiatAmount[];
-      nonNativeTokens: TokenWithFiatAmount[];
-    }>(
-      (acc, token) => {
-        if (token.isNative) {
-          acc.nativeTokens.push(token);
-        } else {
-          acc.nonNativeTokens.push(token);
-        }
-        return acc;
-      },
-      { nativeTokens: [], nonNativeTokens: [] },
-    );
-    const assets = [...nativeTokens, ...nonNativeTokens];
-    return sortAssets(assets, tokenSortConfig);
-  }, [
-    tokenSortConfig,
-    tokenNetworkFilter,
-    conversionRate,
-    contractExchangeRates,
-    currentNetwork,
-    selectedAccount,
-    selectedAccountTokensChains,
-    newTokensImported,
-  ]);
+  const sortedFilteredTokens = useSortedFilteredTokens();
+  const shouldShowFiat = useShouldShowFiat();
```

Comment:
This is a good cleanup! It should make this component easier to test and reason with.